### ISSUE TITLE
Make intern.mode accessible in configuration file

### ISSUE THREAD

**Issue by Blasz** [theintern/intern#263]
With the [existing path issue](https://github.com/theintern/intern/issues/249) between executing tests in the client browser and the runner, some way of distinguishing which mode or environment the test is running in is necessary in the intern configuration for certain config items such as `useLoader['host-browser']`.

The mode is accessible via the intern module _during_ tests as per [the wiki entry](https://github.com/theintern/intern/wiki/Using-Custom-Arguments-and-Configuration-Properties), however, this is currently set to null when accessed in the intern configuration file.

If this is not possible, the path issue needs to be resolved.

In conclusion, if the relative browser paths are not the same depth for both `client.html` (client browser) and `proxyUrl` (runner), any configuration involving browser paths will result in unit tests failing for one of the two environments.

The only work-around I can see at the moment is to pass a commandline argument to both client and runner indicating the mode and then accessing that in the config via `intern.args`, however, I do think this is a hacky work-around.
